test/crypto/sign.js

```diff
@@ -0,0 +1,315 @@
+		const encryptedMessage = encryptMessageWithSecret(
+			secretMessage, defaultSecret, defaultPublicKey,
+		);
+
+		it('should be able to decrypt the message correctly with given receiver secret', () => {
+			const decryptedMessage = decryptMessageWithSecret(
+				encryptedMessage.encryptedMessage, encryptedMessage.nonce, defaultSecret, defaultPublicKey,
+			);
+
+			(decryptedMessage).should.be.ok();
+			(decryptedMessage).should.be.equal(secretMessage);
+		});
+	});
+
+	describe('#signMessageWithTwoSecrets', () => {
+		it('should signTransaction a message using two secrets', () => {
+			const signature = signMessageWithTwoSecrets(
+				notSecretMessage, defaultSecret, defaultSecondSecret,
+			);
+
+			(signature).should.be.equal(defaultTwoSignSignature);
+		});
+	});
+
+	describe('#verifyMessageWithTwoPublicKeys', () => {
+		const publicKey1 = defaultPublicKey;
+		const publicKey2 = defaultSecondPublicKey;
+		const invalidPublicKey1 = 'a4465fd76c16fcc458448076372abf1912cc5b150663a64dffefe550f96fe';
+		const invalidPublicKey2 = 'caf0f4c00cf9240771975e42b6672c88a832f98f01825dda6e001e2aab0bc';
+		it('should verify a message using two publicKeys', () => {
+			const verified = verifyMessageWithTwoPublicKeys(
+				defaultTwoSignSignature, publicKey1, publicKey2,
+			);
+
+			(verified).should.be.equal(notSecretMessage);
+		});
+
+		it('should throw on invalid first publicKey', () => {
+			(function verifyMessageWithFirstInvalidPublicKey() {
+				verifyMessageWithTwoPublicKeys(
+					defaultTwoSignSignature, invalidPublicKey1, publicKey2,
+				);
+			}).should.throw('Invalid first publicKey, expected 32-byte publicKey');
+		});
+
+		it('should throw on invalid second publicKey', () => {
+			(function verifyMessageWithSecondInvalidPublicKey() {
+				verifyMessageWithTwoPublicKeys(
+					defaultTwoSignSignature, publicKey1, invalidPublicKey2,
+				);
+			}).should.throw('Invalid second publicKey, expected 32-byte publicKey');
+		});
+
+		it('should throw on invalid primary signature', () => {
+			(function verifyMessageWithSecondInvalidPublicKey() {
+				const invalidTwoSignSignature = defaultTwoSignSignature.slice(0, 20);
+				verifyMessageWithTwoPublicKeys(
+					invalidTwoSignSignature, publicKey1, publicKey2,
+				);
+			}).should.throw('Invalid signature second publicKey, cannot verify message');
+		});
+
+		it('should throw on invalid secondary signature', () => {
+			(function verifyMessageWithSecondInvalidPublicKey() {
+				const msgBytes = naclInstance.encode_utf8(notSecretMessage);
+				const firstKeys = getRawPrivateAndPublicKeyFromSecret(defaultSecret);
+				const secondKeys = getRawPrivateAndPublicKeyFromSecret(defaultSecondSecret);
+				const signedMessage = naclInstance.crypto_sign(msgBytes, firstKeys.privateKey).slice(0, 20);
+				const doubleSignedMessage = bufferToHex(naclInstance.crypto_sign(
+					signedMessage, secondKeys.privateKey,
+				));
+				verifyMessageWithTwoPublicKeys(doubleSignedMessage, publicKey1, publicKey2);
```

Comment:
Everything except the function under test should be removed from the `should.throw` assertion.